tests/test/java/exceptions.py

```diff
@@ -0,0 +1,228 @@
+                                throw new RuntimeException("ex1");
+                            } catch (final RuntimeException ex) {
+                                throw new RuntimeException("ex2", ex);
+                            }
+                        } catch(final  RuntimeException ex) {
+                            throw new CustomTestException("ex3", ex);
+                        }
+                    }
+                }
+
+                class THROW_EXCEPTION_IN_USER_CODE_WITHIN_REFLECTION {
+                        private static void wrap_custom_test_class() throws ClassNotFoundException, 
+                                                    NoSuchMethodException,
+                                                    InstantiationException, IllegalAccessException,
+                                                    InvocationTargetException {
+                            Class a = Class.forName("com.exasol.CustomTestClass");
+                            Constructor<?> constructor = a.getConstructor();
+                            Object obj = constructor.newInstance();
+                            Method m = a.getMethod("throwEx");
+                            m.invoke(obj);
+                        }
+
+                    public static void run(ExaMetadata exa, ExaIterator ctx) throws Exception {
+                        wrap_custom_test_class();
+                    }
+                }
+                '''))
+        try:
+            self.query("SELECT T1.throw_exception_in_user_code_within_reflection() FROM DUAL")
+        except DataError as ex:
+            exception = ex
+
+        self._verify_exception(exception,
+                              "VM error",
+                              "com.exasol.ExaUDFException: ",
+                              "com.exasol.CustomTestException: ex3",
+                              "com.exasol.CustomTestClass.throwEx(THROW_EXCEPTION_IN_USER_CODE_WITHIN_REFLECTION.java:23)",
+                              "java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)",
+                              "java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke",
+                              "java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke",
+                              "java.base/java.lang.reflect.Method.invoke",
+                              "com.exasol.THROW_EXCEPTION_IN_USER_CODE_WITHIN_REFLECTION.wrap_custom_test_class",
+                              "com.exasol.THROW_EXCEPTION_IN_USER_CODE_WITHIN_REFLECTION.run",
+                              "com.exasol.ExaWrapper.run" )
+
+
+
+    def _verify_exception(self, exception, *args):
+        exception_str = str(exception)
+        exception_str_lines = exception_str.split(sep="\\n")
+        for idx in range(len(args)):
+            if args[idx] not in exception_str_lines[idx]:
+                self.fail(f'"{args[idx]}" not in "{exception_str_lines[idx]}"')
```

Comment:
We only test what should be in the Exception and not what shouldn't be in there

Reply:
Done